test/test_arc.py

```diff
@@ -0,0 +1,112 @@
+import StringIO
+import unittest
+
+from apel.db.records.job import JobRecord
+from apel.parsers.arc import ARCParser
+
+
+class ARCParserTest(unittest.TestCase):
+    def setUp(self):
+        self.parser = ARCParser('testSite', 'testHost', True)
+
+    def test_parse_file(self):
+        """
+        Test that the parser returns a JobRecord and the right number of lines.
+        """
+        arcfile = StringIO.StringIO(complete)
+
+        record, lines = self.parser.parse_arc_file(arcfile)
+        self.assertEqual(type(record), JobRecord)
+        self.assertEqual(lines, 29)
+
+        arcfile.close()
+
+    def test_parse_empty_file(self):
+        """
+        Test that the parser returns None and 0 for an empty file.
+        """
+        empty_file = StringIO.StringIO("")
+
+        record, lines = self.parser.parse_arc_file(empty_file)
+        self.assertEqual(record, None)
+        self.assertEqual(lines, 0)
+
+        empty_file.close()
+
+    def test_invalid_file(self):
+        """
+        Test that the parser raises an exception for an invalid file.
+        """
+        invalid_file = StringIO.StringIO("valid=line\ninvalid line")
+
+        self.assertRaises(ValueError, self.parser.parse_arc_file, invalid_file)
+
+        invalid_file.close()
+
+    def test_parse_incomplete_job(self):
+        """
+        Test that the parser returns None and >0 for an incomplete job.
```

Comment:
Returns `>0`? Not just `0`?

Reply:
Ah, it's not that this function returns 0, but it's checking that another function returns `None` for one value of a tuple and a value greater than 0 for the other. That could be clearer.